cli/src/pcluster/utils.py

```diff
@@ -1076,7 +1076,21 @@ def read_remote_file(url):
         if urlparse(url).scheme == "s3":
             match = re.match(r"s3://(.*?)/(.*)", url)
             bucket, key = match.group(1), match.group(2)
-            file_contents = boto3.resource("s3").Object(bucket, key).get()["Body"].read().decode("utf-8")
+            try:
+                if get_partition() == "aws":
+                    s3_resource = boto3.resource("s3", region_name="us-east-1")
+                else:
+                    s3_resource = boto3.resource("s3")
+                file_contents = s3_resource.Object(bucket, key).get()["Body"].read().decode("utf-8")
+            except ClientError as e:
+                error = e.response["Error"]
+                if error["Code"] == "IllegalLocationConstraintException":
+                    # Try to make the call again if the object is in opt in region.
+                    match = re.match(r"The (.*) location constraint", error["Message"])
```

Comment:
note: the matching word might be `unspecified` or `EU`  

See: 
* https://docs.aws.amazon.com/code-samples/latest/catalog/cpp-s3-create_bucket.cpp.html
* https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketLocation.html#API_GetBucketLocation_ResponseSyntax